pkg/datapath/iptables/iptables.go

```diff
@@ -1585,13 +1584,16 @@ func (m *Manager) installRules(state desiredState) error {
 	return nil
 }
 
-func (m *Manager) remoteSNATDstAddrExclusionCIDR(nativeRoutingCIDR, allocCIDR string) string {
+func (m *Manager) remoteSNATDstAddrExclusionCIDR(nativeRoutingCIDR, allocCIDR string) []string {
+	if len(m.sharedCfg.SnatDstExclusionCIDRs) > 0 {
+		return m.sharedCfg.SnatDstExclusionCIDRs
+	}
 	if nativeRoutingCIDR != "" {
 		// ip{v4,v6}-native-routing-cidr is set, so use it
-		return nativeRoutingCIDR
+		return []string{nativeRoutingCIDR}
```

Comment:
I don't think we should ignore the `nativeRoutingCIDR` or `allocCIDR` CIDRs when `SnatDstExclusionCIDRs` is set (but rather just concat the 2)